src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -655,4 +673,40 @@ where
         self.stream.flush().await?;
         Ok(())
     }
+
+    /// covert_the row to binary format.
+    fn covert_to_binary_format(row: &Row, row_desc: &Vec<PgFieldDescriptor>) -> Vec<Option<Bytes>> {
```

Comment:
I prefer do convert binary to text instead of convert text to binary.

So I think we change `to_pg_rows` to `to_pg_binary_rows()` that return bytes, and convert it back to TEXT if needed. But that requires fully implement ser/desr logic, I'm not sure whether it's easy to do in one PR.

Reply:
BTW it should be convert

Reply:
> I prefer do convert binary to text instead of convert text to binary. Due to the efficiency requirement in BINARY is larger than TEXT.
> 
> So I think we change `to_pg_rows` to `to_pg_binary_rows()` that return bytes, and convert it back to TEXT if needed. But that requires fully implement ser/desr logic, I'm not sure whether it's easy to do in one PR.

I'm agree with you, but I prefer do it in another pr. Because for now 'run_statement' return text format, we need to modify 'run_statement' at the session which involved frontend. It seems too big for this pr.